core/libraries/messages/EE_messenger.lib.php

```diff
@@ -556,7 +556,7 @@ protected function _get_admin_content_events_edit($message_types, $extra)
                 }
                 // if there is no $default_value then we set it as the global
                 $default_value = empty($default_value) ? $mtpgID : $default_value;
-                $edit_url = EEH_URL::add_query_args_and_nonce(array( 'page' => 'espresso_messages', 'action' => 'edit_message_template', 'id' => $default_value ), admin_url('admin.php'));
+                $edit_url = EEH_URL::add_query_args_and_nonce(array( 'page' => 'espresso_messages', 'action' => 'edit_message_template', 'id' => $default_value, 'evt_id' => $event_id ), admin_url('admin.php'));
```

Comment:
might also improve readability here to extract those query vars into a variable
(i know this has not been done so much in a lot of older code, but moving forwards, it is a good practice)

ex:

```php
$query_args = [
	'page' => 'espresso_messages',
	'action' => 'edit_message_template',
	'id' => $default_value,
	'evt_id' => $event_id
];
$edit_url = EEH_URL::add_query_args_and_nonce($query_args, admin_url('admin.php'));
```

Reply:
Kk, done.

I did the create_url code the same way seeing how it was right next to it anyway.